packages/@lwc/template-compiler/src/__tests__/fixtures/directive-lwc-if-else/dynamic/if/expected.js

```diff
@@ -1,13 +1,18 @@
 import { registerTemplate } from "lwc";
 const stc0 = {
-  key: 0,
+  key: 1,
 };
-const stc1 = [];
 function tmpl($api, $cmp, $slotset, $ctx) {
-  const { dc: api_dynamic_component } = $api;
-  return $cmp.visible.if
-    ? [api_dynamic_component("x-foo", $cmp.trackedProp.foo, stc0)]
-    : stc1;
+  const { dc: api_dynamic_component, fr: api_fragment } = $api;
+  return [
+    $cmp.visible.if
+      ? api_fragment(
+          0,
+          [api_dynamic_component("x-foo", $cmp.trackedProp.foo, stc0)],
+          0
+        )
+      : null,
```

Comment:
@jodarove This now returns null if there's nothing in the else branch

Reply:
I think this should be ok, it looks like there's a [check](https://github.com/salesforce/lwc/blob/11d6de34934b9f984f2372a7d947a753d29a5e31/packages/%40lwc/engine-core/src/framework/rendering.ts#L381-L395) when mounting the vnodes to see if the value is null.

We also return an [array of null values](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/template-compiler/src/__tests__/fixtures/directive-if/template-if-else/expected.js#L6-L9) in the `if:true/false` cases from the tmpl function.

Reply:
Yup. I modeled this after codegen's [behavior](https://github.com/salesforce/lwc/blob/master/packages/@lwc/template-compiler/src/codegen/index.ts#L225) for if:true/false.